raster-layer-file/src/main/java/com/esri/arcgisruntime/sample/rasterlayerfile/MainActivity.java

```diff
@@ -0,0 +1,116 @@
+/* Copyright 2017 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.rasterlayerfile;
+
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.Toast;
+
+import com.esri.arcgisruntime.layers.RasterLayer;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.raster.Raster;
+
+import java.io.File;
+
+/**
+ * A sample class which demonstrates loading a Raster from the local device.
+ */
+public class MainActivity extends AppCompatActivity {
+
+    private MapView mMapView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        // retrieve the MapView from layout
+        mMapView = (MapView) findViewById(R.id.mapView);
+
+        // define permission to request
+        String[] reqPermission = new String[] { Manifest.permission.WRITE_EXTERNAL_STORAGE };
+        int requestCode = 2;
+        // For API level 23+ request permission at runtime
+        if(ContextCompat.checkSelfPermission(MainActivity.this,
+                reqPermission[0]) == PackageManager.PERMISSION_GRANTED){
+            loadRaster();
+        } else {
+            // request permission
+            ActivityCompat.requestPermissions(MainActivity.this, reqPermission, requestCode);
+        }
+    }
+
+    /**
+     * Loads Shasta.tif as a Raster and adds it to a new RasterLayer. The RasterLayer is then added
+     * to the map as an operational layer. Map viewpoint is then set based on the raster's geometry.
+     */
+    private void loadRaster() {
+        // create a raster from a local raster file
+        Raster raster = new Raster(
+                new File("/storage/emulated/0/ArcGIS/samples/raster/Shasta.tif").getAbsolutePath());
```

Comment:
Avoid hard coding the path, use `Environment.getExternalStorageDirectory()` and the local path as a string resource.